sqlglot/helper.py

```diff
@@ -432,24 +432,28 @@ def first(it: t.Iterable[T]) -> T:
 
 def case_sensitive(text: str, dialect: DialectType) -> bool:
     """Checks if text contains any case sensitive characters depending on dialect."""
-    from sqlglot.dialects.dialect import RESOLVES_IDENTIFIERS_AS_UPPERCASE
+    from sqlglot.dialects.dialect import Dialect
 
-    unsafe = str.islower if dialect in RESOLVES_IDENTIFIERS_AS_UPPERCASE else str.isupper
+    dialect = Dialect.get_or_raise(dialect)
+    if dialect.RESOLVES_IDENTIFIERS_AS_UPPERCASE is None:
+        return False
```

Comment:
Note: maybe we can refactor this section somehow because we're doing a `Dialect` lookup every time, so the overhead may be non-trivial when we're calling `case_sensitive` in `_quote` which is passed to `transform` (inside of `quote_identifiers`). Similarly, `should_identify` calls `case_sensitive` as well.

Reply:
Good point -- also moved `_quote` in `Dialect` for the same reason.